src/main/java/com/hierynomus/msdfsc/ReferralCache.java

```diff
@@ -183,4 +170,46 @@ public String toString() {
 
     }
 
+
+    static class ReferralCacheNode {
+        String pathComponent;
+        Map<String, ReferralCacheNode> childNodes = new HashMap<>();
```

Comment:
Use ConcurrentHashMap to avoid potential ConcurrentModificationexceptions?

Reply:
Yes, indeed.